store/driver/txn/error.go

```diff
@@ -158,33 +158,45 @@ func newWriteConflictError(conflict *kvrpcpb.WriteConflict) error {
 	if conflict == nil {
 		return kv.ErrWriteConflict
 	}
-	var buf bytes.Buffer
-	prettyWriteKey(&buf, conflict.Key)
-	buf.WriteString(" primary=")
-	prettyWriteKey(&buf, conflict.Primary)
-	return kv.ErrWriteConflict.FastGenByArgs(conflict.StartTs, conflict.ConflictTs, conflict.ConflictCommitTs, buf.String(), conflict.Reason.String())
+	var buf1 bytes.Buffer // table id part of conflict key, which is used to be parsed by upper level to provide more information about the table
+	var buf2 bytes.Buffer // the rest part of conflict key
+	var buf3 bytes.Buffer // table id part of primary key
+	var buf4 bytes.Buffer // the rest part of primary key
+	prettyWriteKey(&buf1, &buf2, conflict.Key)
+	buf2.WriteString(", primary=")
+	prettyWriteKey(&buf3, &buf4, conflict.Primary)
+	return kv.ErrWriteConflict.FastGenByArgs(conflict.StartTs, conflict.ConflictTs, conflict.ConflictCommitTs,
+		buf1.String(), buf2.String(), buf3.String(), buf4.String(), conflict.Reason.String())
 }
 
-func prettyWriteKey(buf *bytes.Buffer, key []byte) {
+func prettyWriteKey(buf1, buf2 *bytes.Buffer, key []byte) {
```

Comment:
It would be confusing to use `buf1`, `buf2` in a function signature, how about changing them to `bufTableID`, `bufIndexID` etc?

Reply:
OK